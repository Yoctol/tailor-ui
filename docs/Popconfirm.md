---
id: popconfirm
title: Popconfirm
---

A simple and compact confirmation dialog of an action.

## When To Use

A simple and compact dialog used for asking for user confirmation.

## Examples

```js
import { Popconfirm } from 'tailor-ui';
```

### Basic

```jsx live
<Popconfirm
  position={Position.RIGHT}
  content="Are you sure to delete this todo?"
  onConfirm={() => console.log('confirm')}
  onCancel={() => console.log('cancel')}
>
  <Button>Button</Button>
</Popconfirm>
```

### With customize text & type

```jsx live
() => {
  const modal = useModal();
  const message = useMessage();

  return (
    <Popconfirm
      position={Position.RIGHT}
      type="error"
      confirmText="毀了它"
      cancelText="一小時後再提醒我"
      content="此操作將會毀滅本系統，你確定？"
      onConfirm={() => {
        console.log('confirm');
        message.success('您的系統已被毀滅');
      }}
      onCancel={() => {
        modal.success({
          title: '我才不會提醒你',
          content: '您的系統已被毀滅',
          confirmText: '感謝',
        });
        console.log('cancel');
      }}
    >
      <Button>Button</Button>
    </Popconfirm>
  );
}
```

## API

| Property          | Description                                    | Type                                                | Default     |
|-------------------|------------------------------------------------|-----------------------------------------------------|-------------|
| `content`         | content of the confirmation box                | `ReactNode`                                         |             |
| `type`            | customize icon of confirmation                 | `'info'` \| `'success'` \| `'error'` \| `'warning'` | `'warning'` |
| `cancelText`      | text of the Cancel button                      | `string`                                            |    Cancel         |
| `confirmText`     | text of the Confirm button                     | `string`                                            |     Confirm        |
| `onCancel`        | callback of cancel                             | `(event: MouseEvent) => void`                       |             |
| `onConfirm`       | callback of confirmation                       | `(event: MouseEvent) => void`                       |             |

Consult [Popover's documentation](Popover.md#api) to find more APIs.
