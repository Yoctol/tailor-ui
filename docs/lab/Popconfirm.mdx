---
name: Popconfirm
menu: Lab
route: /lab/popconfirm
---

import { PropsTable, Playground } from 'docz';
import { Button, Position, message, useModal } from 'tailor-ui';
import { Popconfirm } from '@tailor-ui/lab';

# Popconfirm

A simple and compact confirmation dialog of an action.

## When To Use

A simple and compact dialog used for asking for user confirmation.

## Examples

```js
import { Popconfirm } from 'tailor-ui';
```

### Basic

<Playground>
  <Popconfirm
    position={Position.RIGHT}
    content="Are you sure to delete this todo?"
    onConfirm={() => console.log('confirm')}
    onCancel={() => console.log('cancel')}
  >
    <Button>Button</Button>
  </Popconfirm>
</Playground>

### With customize text & type

<Playground>
  {() => {
    const modal = useModal();

    return (
      <Popconfirm
        position={Position.RIGHT}
        type="error"
        confirmText="毀了它"
        cancelText="一小時後再提醒我"
        content="此操作將會毀滅本系統，你確定？"
        onConfirm={() => {
          console.log('confirm');
          message.success('您的系統已被毀滅');
        }}
        onCancel={() => {
          modal.success({
            title: '我才不會提醒你',
            content: '您的系統已被毀滅',
            confirmText: '感謝',
          });
          console.log('cancel');
        }}
      >
        <Button>Button</Button>
      </Popconfirm>
    )

}}

</Playground>

## API

<PropsTable of={Popconfirm} />
