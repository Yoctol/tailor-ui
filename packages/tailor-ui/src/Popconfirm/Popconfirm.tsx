import React, {
  FunctionComponent,
  MouseEvent,
  ReactNode,
  memo,
  useContext,
} from 'react';

import Button from '../Button';
import Popover, { PopoverProps } from '../Popover';
import getTypeIcon from '../utils/getTypeIcon';
import { Flex } from '../Layout';
import { LocaleContext } from '../UIProvider';

interface PopconfirmContentProps extends PopoverProps {
  /**
   * text of the Cancel button
   */
  cancelText?: string;
  /**
   * text of the Confirm button
   */
  confirmText?: string;
  /**
   * content of the confirmation box
   */
  content: ReactNode;
  /**
   * customize icon of confirmation
   */
  type?: 'info' | 'success' | 'warning' | 'error';
  /**
   * callback of cancel
   */
  onCancel?: (event: MouseEvent) => void;
  /**
   * callback of confirmation
   */
  onConfirm?: (event: MouseEvent) => void;
  onOpenComplete?: () => void;
  onCloseComplete?: () => void;
}

const PopconfirmContent: FunctionComponent<
  PopconfirmContentProps & {
    hideTooltip: () => void;
  }
> = memo(function PopconfirmContent({
  type,
  content,
  confirmText,
  cancelText,
  onConfirm,
  onCancel,
  hideTooltip,
}) {
  const { locale } = useContext(LocaleContext);

  const icon = getTypeIcon(type);

  return (
    <>
      <Flex alignItems="center">
        {icon}
        {content}
      </Flex>
      <Flex mt="3">
        <Button
          ml="auto"
          size="sm"
          onClick={(event: MouseEvent) => {
            hideTooltip();
            if (onCancel) {
              onCancel(event);
            }
          }}
        >
          {cancelText || locale.Popconfirm.cancelText}
        </Button>
        <Button
          ml="2"
          size="sm"
          variant="primary"
          onClick={(event: MouseEvent) => {
            hideTooltip();
            if (onConfirm) {
              onConfirm(event);
            }
          }}
        >
          {confirmText || locale.Popconfirm.confirmText}
        </Button>
      </Flex>
    </>
  );
});

const Popconfirm: FunctionComponent<PopconfirmContentProps> = ({
  type,
  content,
  confirmText,
  cancelText,
  onConfirm,
  onCancel,
  ...props
}) => (
  <Popover
    content={hideTooltip => (
      <PopconfirmContent
        type={type}
        content={content}
        confirmText={confirmText}
        cancelText={cancelText}
        onConfirm={onConfirm}
        onCancel={onCancel}
        hideTooltip={hideTooltip}
      />
    )}
    {...props}
  />
);

Popconfirm.defaultProps = {
  content: '',
  type: 'warning',
  onConfirm: () => {},
  onCancel: () => {},
};

export default Popconfirm;
