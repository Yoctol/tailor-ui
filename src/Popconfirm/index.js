import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { themeGet } from 'styled-system';

import Button from '../Button';
import Flex from '../Grid/Flex';
import Space from '../Grid/Space';
import Tooltip from '../Tooltip';
import getTypeIcon from '../utils/getTypeIcon';
import { ArrowComponent, ContentComponent } from '../Popover';

const components = {
  ArrowComponent,
  ContentComponent,
};

const StyledContent = styled(Space)`
  padding: ${themeGet('space.2')};
  word-break: break-all;
  white-space: nowrap;
`;

const Popconfirm = ({
  confirmText,
  cancelText,
  onConfirm,
  onCancel,
  type,
  content,
  ...props
}) => {
  const icon = getTypeIcon(type);
  const renderContent = hideTooltip => (
    <StyledContent>
      <Flex alignItems="center">
        {icon}
        {content}
      </Flex>
      <Flex mt="3">
        <Button
          ml="auto"
          size="sm"
          onClick={event => {
            hideTooltip();
            onCancel(event);
          }}
        >
          {cancelText}
        </Button>
        <Button
          ml="2"
          size="sm"
          type="primary"
          onClick={event => {
            hideTooltip();
            onConfirm(event);
          }}
        >
          {confirmText}
        </Button>
      </Flex>
    </StyledContent>
  );

  return (
    <Tooltip
      components={components}
      content={renderContent}
      trigger="click"
      {...props}
    />
  );
};

Popconfirm.propTypes = {
  /**
   * text of the Cancel button
   */
  cancelText: PropTypes.string,
  /**
   * text of the Confirm button
   */
  confirmText: PropTypes.string,
  /**
   * content of the confirmation box
   */
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /**
   * customize icon of confirmation
   */
  type: PropTypes.oneOf(['info', 'success', 'warning', 'error']),
  /**
   * callback of cancel
   */
  onCancel: PropTypes.func,
  /**
   * callback of confirmation
   */
  onConfirm: PropTypes.func,
};

Popconfirm.defaultProps = {
  confirmText: 'Confirm',
  cancelText: 'Cancel',
  content: '',
  type: 'warning',
  onConfirm: () => {},
  onCancel: () => {},
};

export default Popconfirm;
