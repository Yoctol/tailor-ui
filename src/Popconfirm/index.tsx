import React, { MouseEvent, PureComponent, ReactNode } from 'react';
import { omit } from 'ramda';

import styled from 'utils/styled-components';

import Button from '../Button';
import Flex from '../Grid/Flex';
import Space from '../Grid/Space';
import Tooltip, { ITooltipProps } from '../Tooltip';
import getTypeIcon from '../utils/getTypeIcon';
import { ArrowComponent, ContentComponent } from '../Popover';
import { LocaleConsumer } from '../UIProvider';

const components = {
  ArrowComponent,
  ContentComponent,
};

const StyledContent = styled(Space)`
  padding: ${p => p.theme.space[2]};
  word-break: break-all;
  white-space: nowrap;
`;

export type PopconfirmProps = ITooltipProps & {
  /**
   * text of the Cancel button
   */
  cancelText?: string;
  /**
   * text of the Confirm button
   */
  confirmText?: string;
  /**
   * content of the confirmation box
   */
  content: ReactNode;
  /**
   * customize icon of confirmation
   */
  type?: 'info' | 'success' | 'warning' | 'error';
  /**
   * callback of cancel
   */
  onCancel?: (event: MouseEvent) => void;
  /**
   * callback of confirmation
   */
  onConfirm?: (event: MouseEvent) => void;
};

class Popconfirm extends PureComponent<PopconfirmProps> {
  static defaultProps = {
    content: '',
    type: 'warning',
    onConfirm: () => {},
    onCancel: () => {},
  };

  renderContent = (hideTooltip: () => void) => {
    const {
      type = 'warning',
      content = '',
      confirmText,
      cancelText,
      onConfirm,
      onCancel,
    } = this.props;

    const icon = getTypeIcon(type);

    return (
      <LocaleConsumer>
        {({ locale }) => (
          <StyledContent>
            <Flex alignItems="center">
              {icon}
              {content}
            </Flex>
            <Flex mt="3">
              <Button
                ml="auto"
                size="sm"
                onClick={(event: MouseEvent) => {
                  hideTooltip();
                  if (onCancel) {
                    onCancel(event);
                  }
                }}
              >
                {cancelText || locale.Popconfirm.cancelText}
              </Button>
              <Button
                ml="2"
                size="sm"
                type="primary"
                onClick={(event: MouseEvent) => {
                  hideTooltip();
                  if (onConfirm) {
                    onConfirm(event);
                  }
                }}
              >
                {confirmText || locale.Popconfirm.confirmText}
              </Button>
            </Flex>
          </StyledContent>
        )}
      </LocaleConsumer>
    );
  };

  render() {
    const props = omit(
      ['type', 'content', 'confirmText', 'cancelText', 'onConfirm', 'onCancel'],
      this.props
    );

    return (
      <Tooltip
        components={components}
        content={this.renderContent}
        trigger="click"
        {...props}
      />
    );
  }
}

export default Popconfirm;
