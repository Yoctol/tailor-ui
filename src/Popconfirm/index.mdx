---
name: Popconfirm
menu: Components
route: /components/popconfirm
---

import { PropsTable, Playground, Link } from 'docz';

import { Button, Modal, message } from '../';

import Popconfirm from './';

# Popconfirm

A simple and compact confirmation dialog of an action.

## When To Use

A simple and compact dialog used for asking for user confirmation.

## Examples

```js
import { Popconfirm } from '@yoctol/ui';
```

### Basic

<Playground>
  <Popconfirm
    placement="right"
    content="Are you sure to delete this todo?"
    onConfirm={() => console.log('confirm')}
    onCancel={() => console.log('cancel')}
  >
    <Button>Button</Button>
  </Popconfirm>
</Playground>

### With customize text & type

<Playground>
  <Popconfirm
    placement="right"
    type="error"
    confirmText="毀了它"
    cancelText="一小時後再提醒我"
    content="此操作將會毀滅本系統，你確定？"
    onConfirm={() => {
      console.log('confirm');
      message.success('您的系統已被毀滅');
    }}
    onCancel={() => {
      Modal.success({
        title: '我才不會提醒你',
        content: '您的系統已被毀滅',
        confirmText: '感謝',
      });
      console.log('cancel');
    }}
  >
    <Button>Button</Button>
  </Popconfirm>
</Playground>

## API

<PropsTable of={Popconfirm} />

<div>
  Consult{' '}
  <Link to="/data-display/tooltip#properties"> Tooltip's documentation</Link> to
  find more APIs.
</div>
