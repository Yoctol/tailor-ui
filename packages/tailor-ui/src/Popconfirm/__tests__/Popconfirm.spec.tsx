import React from 'react';

import { fireEvent, mockRaf, render, useMockRaf } from 'test/test-utils';

import { Popconfirm } from '../Popconfirm';

describe('Popconfirm', () => {
  useMockRaf();

  it('should render correctly', () => {
    const { baseElement } = render(
      <Popconfirm visible content="Content">
        <button type="button">button</button>
      </Popconfirm>
    );

    mockRaf.flushSpring();

    expect(baseElement).toMatchSnapshot();
  });

  it('should trigger onConfirm correctly', () => {
    const confirmFn = jest.fn();

    const { getByText } = render(
      <Popconfirm visible content="Content" onConfirm={confirmFn}>
        <button type="button">button</button>
      </Popconfirm>
    );

    mockRaf.flushSpring();

    const confirmButton = getByText('Confirm');
    fireEvent.click(confirmButton);

    expect(confirmFn).toBeCalled();
  });

  it('should trigger onCancel correctly', () => {
    const cancelFn = jest.fn();

    const { getByText } = render(
      <Popconfirm visible content="Content" onCancel={cancelFn}>
        <button type="button">button</button>
      </Popconfirm>
    );

    mockRaf.flushSpring();

    const cancelButton = getByText('Cancel');
    fireEvent.click(cancelButton);

    expect(cancelFn).toBeCalled();
  });
});
