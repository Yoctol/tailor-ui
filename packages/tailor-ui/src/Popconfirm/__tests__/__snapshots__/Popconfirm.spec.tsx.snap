// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm should render correctly 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  cursor: default;
  margin-right: 0.5rem;
}

.c5 svg {
  fill: #8b8b8b;
  vertical-align: middle;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c5.c5.c5 svg {
  fill: #f6a700;
  width: 20px;
  height: 20px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid;
  border-radius: 0.5rem;
  border-color: transparent;
  line-height: 1.5;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 1rem;
  border-color: #e0e0e0;
  background-color: #ffffff;
  color: #373737;
}

.c0:focus {
  outline: 0;
}

.c0:disabled,
.c0[disabled] {
  opacity: 0.85;
  background-color: #e0e0e0;
  color: #8b8b8b;
  cursor: default;
  pointer-events: none;
}

.c0:disabled .c4 svg,
.c0[disabled] .c4 svg {
  fill: #8b8b8b;
}

.c0::before {
  content: '';
  display: inline-block;
  position: relative;
  left: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 0;
  height: 0;
  margin-right: 0;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c0::before {
  border-color: #373737;
}

.c0 .c4 svg {
  fill: #bfbfbf;
}

.c0:hover {
  border-color: #6B88c5;
}

.c0:active {
  border-color: #3c5ad0;
  color: #3c5ad0;
}

.c0:active .c4 svg {
  fill: #3c5ad0;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid;
  border-radius: 0.5rem;
  border-color: transparent;
  line-height: 1.5;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
  height: 1.75rem;
  padding: 0 0.5rem;
  font-size: 0.875rem;
  border-color: #e0e0e0;
  background-color: #ffffff;
  color: #373737;
  margin-left: auto;
}

.c7:focus {
  outline: 0;
}

.c7:disabled,
.c7[disabled] {
  opacity: 0.85;
  background-color: #e0e0e0;
  color: #8b8b8b;
  cursor: default;
  pointer-events: none;
}

.c7:disabled .c4 svg,
.c7[disabled] .c4 svg {
  fill: #8b8b8b;
}

.c7::before {
  content: '';
  display: inline-block;
  position: relative;
  left: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 0;
  height: 0;
  margin-right: 0;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c7::before {
  border-color: #373737;
}

.c7 .c4 svg {
  fill: #bfbfbf;
}

.c7:hover {
  border-color: #6B88c5;
}

.c7:active {
  border-color: #3c5ad0;
  color: #3c5ad0;
}

.c7:active .c4 svg {
  fill: #3c5ad0;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid;
  border-radius: 0.5rem;
  border-color: transparent;
  line-height: 1.5;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
  height: 1.75rem;
  padding: 0 0.5rem;
  font-size: 0.875rem;
  background-color: #3c5ad0;
  color: #ffffff;
  font-weight: bold;
  margin-left: 0.5rem;
}

.c8:focus {
  outline: 0;
}

.c8:disabled,
.c8[disabled] {
  opacity: 0.85;
  background-color: #e0e0e0;
  color: #8b8b8b;
  cursor: default;
  pointer-events: none;
}

.c8:disabled .c4 svg,
.c8[disabled] .c4 svg {
  fill: #8b8b8b;
}

.c8::before {
  content: '';
  display: inline-block;
  position: relative;
  left: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 0;
  height: 0;
  margin-right: 0;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c8 .c4 svg {
  fill: #ffffff;
}

.c8:hover {
  background-color: #516bd5;
}

.c8:active {
  background-color: #2f4dc3;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  box-sizing: border-box;
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  position: absolute;
  width: 0;
  height: 0;
  border-top: 5px solid #001e40;
  border-right: 5px solid transparent;
  border-left: 5px solid transparent;
  opacity: 0.8;
  top: -5px;
  left: 50%;
  margin-left: -3px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  border-top: 5px solid #e0e0e0;
  opacity: 1;
}

.c9::after {
  content: '';
  position: absolute;
  top: -7px;
  left: -6px;
  border-top: 6px solid #ffffff;
  border-right: 6px solid transparent;
  border-left: 6px solid transparent;
}

.c1 {
  border: 1px solid;
  border-color: #001e40;
  background-color: #001e40;
  box-shadow: 0 2px 8px rgba(0,0,0,0.1);
  color: #ffffff;
  font-size: 0.875rem;
  white-space: nowrap;
  opacity: 0.8;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  border-radius: 0.25rem;
  text-align: center;
  padding: 0;
  border-color: #e0e0e0;
  opacity: 1;
  background-color: #ffffff;
  color: #373737;
  text-align: left;
}

.c2 {
  padding: 0.5rem;
  word-break: break-all;
  white-space: nowrap;
}

<body>
  <div>
    <button
      class="c0"
    >
      Button
    </button>
  </div>
  <div
    portal-container=""
  >
    <div
      style="position: fixed; top: 0px; left: 0px; z-index: 10;"
    >
      <div
        style="position: absolute; top: 0px; left: 0px; transform: translate3d(0px, 10px, 0px);"
      >
        <div
          style="opacity: 0; transform: scale(0.8);"
        >
          <div
            class="c1"
          >
            <div
              class="c2"
            >
              <div
                class="c3"
              >
                <i
                  class="c4 c5"
                  cursor="default"
                  fill="warning"
                  size="20"
                >
                  <svg
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20.566 3.47l7.964 7.964c1.587 1.588 1.998 2.329 2.268 3.22.27.89.27 1.801 0 2.692-.27.891-.68 1.632-2.268 3.22l-7.964 7.964c-1.588 1.587-2.329 1.998-3.22 2.268-.89.27-1.801.27-2.692 0-.891-.27-1.632-.68-3.22-2.268L3.47 20.566c-1.587-1.588-1.998-2.329-2.268-3.22a4.577 4.577 0 0 1 0-2.692c.27-.891.68-1.632 2.268-3.22l7.964-7.964c1.588-1.587 2.329-1.998 3.22-2.268.89-.27 1.801-.27 2.692 0 .891.27 1.632.68 3.22 2.268zM14 7v10.215c0 .503.07.773.2 1.018.131.244.323.436.567.567.245.13.515.2 1.018.2h.43c.503 0 .773-.07 1.018-.2.244-.131.436-.323.567-.567.13-.245.2-.515.2-1.018V7h-4zm1.785 14c-.503 0-.773.07-1.018.2-.244.131-.436.323-.567.567-.13.245-.2.515-.2 1.018v.43c0 .503.07.773.2 1.018.131.244.323.436.567.567.245.13.515.2 1.018.2h.43c.503 0 .773-.07 1.018-.2.244-.131.436-.323.567-.567.13-.245.2-.515.2-1.018v-.43c0-.503-.07-.773-.2-1.018a1.363 1.363 0 0 0-.567-.567c-.245-.13-.515-.2-1.018-.2h-.43z"
                      fill-rule="nonzero"
                    />
                  </svg>
                </i>
                Test Popconfirm
              </div>
              <div
                class="c6"
              >
                <button
                  class="c7"
                >
                  Cancel
                </button>
                <button
                  class="c8"
                >
                  Confirm
                </button>
              </div>
            </div>
          </div>
          <div
            class="c9"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
